remote/remote_state_s3.go

```diff
@@ -663,6 +944,39 @@ func EnableEnforcedTLSAccesstoS3Bucket(s3Client *s3.S3, config *ExtendedRemoteSt
 	return nil
 }
 
+// Helper function to check if the enforced TLS policy is enabled for the bucket
+func checkIfBucketEnforcedTLS(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) (bool, error) {
+	terragruntOptions.Logger.Debugf("Checking if bucket %s is enforced with TLS", config.Bucket)
+
+	policyOutput, err := s3Client.GetBucketPolicy(&s3.GetBucketPolicyInput{
+		Bucket: aws.String(config.Bucket),
+	})
+	if err != nil {
+		terragruntOptions.Logger.Debugf("Could not get policy for bucket %s", config.Bucket)
+		return false, nil
```

Comment:
`err` is not returned, it is by intention?

Reply:
Yes, but I improved this error check to return no err when bucket dont have policy